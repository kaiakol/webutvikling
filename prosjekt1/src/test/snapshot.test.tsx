import { render } from "@testing-library/react";
import Home from "../pages/home/home";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { expect, it } from "vitest";

it("renders correctly", () => {
  const queryClient = new QueryClient();
  const home = (
    <QueryClientProvider client={queryClient}>
      <Home />
    </QueryClientProvider>
  );
  const rh = render(home);
  expect(rh).toMatchSnapshot();
});
