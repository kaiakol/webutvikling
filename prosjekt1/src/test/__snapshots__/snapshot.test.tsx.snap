// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel"
      >
        <div
          class="carousel-actions"
          id="button-prev-container"
        >
          <button
            aria-label="Previous"
            id="carousel-button-prev"
          >
            &lt;
          </button>
        </div>
        <div>
          Loading ...
        </div>
        <div
          class="carousel-actions"
          id="button-next-container"
        >
          <button
            aria-label="Next"
            id="carousel-button-next"
          >
            &gt;
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel"
    >
      <div
        class="carousel-actions"
        id="button-prev-container"
      >
        <button
          aria-label="Previous"
          id="carousel-button-prev"
        >
          &lt;
        </button>
      </div>
      <div>
        Loading ...
      </div>
      <div
        class="carousel-actions"
        id="button-next-container"
      >
        <button
          aria-label="Next"
          id="carousel-button-next"
        >
          &gt;
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
